### DISCUSSION TITLE
Benchmarking and Performance

### DISCUSSION THREAD

**Discussion by Paramjeet2810** [enisdenjo/graphql-ws#165] [Q&A] [answered]
Hey, So I am planning to use this graphql subscription tool for a chat like application. I want to know if there is a safety cap on the number of graphql subscriptions that should be made per page. 

Is there documentation explaining benchmarking of this tool?

**Comment by enisdenjo** [comment, marked as answer]
Hey there! Happy you're considering this library. Sadly, I havent really benchmarked `graphql-ws`, my free time is scarce. But I can tell you that it is true zero dependency and leverages only JavaScript primitives, so your real cap is memory and CPU power. Or did you have some other "safety cap" in mind?

**Reply by Paramjeet2810** [reply]
On the client-side(frontend), for a regular laptop how many maximum subscriptions would give a nice user experience

**Reply by enisdenjo** [reply]
I really cant give you an exact number. It depends on a number of factors, from internet stability up to the laptop's resources. It also depends on what you mean by "nice user experience" - no subscription message will be swallowed by the lib, no matter the pressure.

WebSockets are designed to handle lots of messages and are often used for real-time games. The library itself will handle any size; but, it also depends on how **you** handle those messages and what you do with them (how many DOM updates, how many resource intensive calculations you perform on each message, etc.).

The only way to really know is to benchmark the library and use the upper median. I welcome anyone to do it, but I really dont have the time for it right now.
